scheduler/src/cook/rest/api.clj

```diff
@@ -1420,9 +1430,22 @@
 
 (def leader-hostname-regex #"^([^#]*)#([0-9]*)#([a-z]*)#.*")
 
-(defn leader-url
+(defn leader-selector->leader-id
+  "Get the current leader node's id from a leader-selector object.
+   Throws if there is currently no leader available."
+  [^LeaderSelector leader-selector]
+  (let [leader (.getLeader leader-selector)]
+    ;; NOTE: .getLeader returns a dummy object when no leader is available,
+    ;; but the dummy object always returns false for the .isLeader predicate.
+    (when-not (.isLeader leader)
+      (throw (IllegalStateException. "Leader is temporarily unavailable.")))
```

Comment:
Are previous consumers of `leader-selector->leader-url` affected by this new exception?

Reply:
No, the original function that calls this function (i.e., `leader-selector->leader-url`) already throws `IllegalStateException` when the leader id is in an invalid format, which is what happens when there's no leader and the id comes back blank. The only thing that changed is that we get a more specific error message now.